### ISSUE TITLE
User is able to send and receive $ 0.000 amount by tapping on the “Max” button from the amount field of “Withdraw CELO” screen while withdrawing CELO. 

### ISSUE THREAD

**Issue by Lss-Ankit** [celo-org/celo-monorepo#4879]
**Frequency:** 100%
**Repro App version:**  Test Flight build Valora app build  v1.0.1 (17) , Android Valora Mainnet build v1.0.0 (16)
**Repro on device:** iPhone 7+ (13.3.1), Google Pixel Xl (8.1) 
**Test Account:** Backup Key: (757) 699-9899. 
Pre-condition: User has buy and Sold  same number CELO. 
**Repro Steps:**
1. Launch the Application and Sign in with test account. 
2. Purchase .04 CELO. 
3. Sold .04 CELO. 
4. Tap on Withdraw CELO section. 
5. Scan other QR code. 
6. Tap on Max button on Amount CELO field. 
7. Tap on Review and then Withdraw button . 
8. Observe. 

**Current Behavior:** User is able to send and receive $ 0.000 amount CELO. 
**Expected Behavior:**  User should not be able to send / receive $0.0 CELO. 
**Investigation:** Issue does repro when user enter amount 0.00000001
**Attachment:** [0_AmountWithdrawCELO.mp4](https://drive.google.com/file/d/1u86SiK181mpc3aaOYF1yze_T7g1NHVhi/view?usp=sharing) (Bug observe at 26 sec)

**Comment by tarikbellamine** [comment]
@nityas curious what you think we should do here. I think we could do 1 of 2 things:
1. Allow users to withdraw "dust" if they so wish. This would require us to change our sig figs on display so that it doesn't show up as "0.000". Maybe scientific notation once it hits 3 decimal points?
2. Impose a minimum withdrawal limit that aligns with our current display minimum. This would probably require us to show the user's balance as 0 when they have less than a certain amount.

**Comment by Lss-Ankit** [comment]
Hi @tarikbellamine I have verified this issue using latest Android internal build v1.3.0 (1004294316) & Test Flight build v1.3.0(27) and observed that user gets result like "9e-7" when user taps on 'Max' button of 'Amount(CELO)' field on 'Withdraw CELO' page while user having less CELO balance (i.e. less than 0.00).
So is it expected???

**Comment by gnardini** [comment]
It should probably not show up in scientific notation, that needs to get fixed.

I support the second proposal from @tarikbellamine anyways, just add a minimum amount to withdraw of 0.001 maybe? Since we show 3 decimals precision for CELO.

**Comment by tarikbellamine** [comment]
Yeah and I think there is currently an open bug because we are using scientific notation somewhere in the exchange flow IIRC. I also think this may be a subset of another issue...I don't remember exactly but I think there was one about displaying minimum digits.

**Comment by Lss-Ankit** [comment]
Hi @tarikbellamine @gnardini I have verified this issue using latest Android Internal Build v1.6.0(1004294325) & Test Flight build v1.6.0(36) & observed that issue is fixed now & User not able to do transaction if user has <$0.00 balance.
**Repro on devices:** Samsung Galaxy A5(7.0) ,iPhone SE (13.5.1)

**Comment by Celoqa** [comment]
Hi @tarikbellamine @gnardini I have verified this issue using latest Android Internal Build v1.6.0(1004294325) & Test Flight build v1.6.0(36) & observed that issue is fixed now & Review button not enabled & hence user not able to transaction if user has 0.00 CELO balance.
**Repro on devices:** Samsung Galaxy A5(7.0) ,iPhone SE (13.4.1)
